Mathlib/CategoryTheory/Limits/Shapes/Reflexive.lean

```diff
@@ -148,6 +171,372 @@ instance (priority := 100) hasCoreflexiveEqualizers_of_hasEqualizers [HasEqualiz
+             have hrᵣ : rᵣ ∈ p := by
+              · suffices f : e₀ ⟶  rᵣ by exact (h₂ f).mp h₁
+                exact StructuredArrow.homMk WalkingParallelPairHom.right
+             cases y <;> cases f
+             rotate_right 3
+             · exact h₁
+             · exact hrₗ
+             · exact hrᵣ
+             · set v : StructuredArrow WalkingReflexivePair.one inclusionWalkingReflexivePair :=
+                StructuredArrow.mk (Y := WalkingParallelPair.zero) left_reflexion
+               suffices f : v ⟶  rₗ by exact (h₂ f).mpr hrₗ
+               exact StructuredArrow.homMk (WalkingParallelPairHom.left)
+             · set v : StructuredArrow WalkingReflexivePair.one inclusionWalkingReflexivePair :=
+                StructuredArrow.mk (Y := WalkingParallelPair.zero) right_reflexion
+               suffices f : v ⟶  rᵣ by exact (h₂ f).mpr hrᵣ
+               exact StructuredArrow.homMk (WalkingParallelPairHom.right)
+
+end WalkingParallelPair
+
+end Limits
+
+namespace Limits
+
+open WalkingReflexivePair
+
+open WalkingReflexivePairHom
+
+variable {C : Type u} [Category.{v} C]
+
+variable {A B : C}
+
+/-- Bundle the data of a parallel pair along with a common section as a functor out of the walking
+reflexive pair -/
+def reflexivePair (f g : A ⟶ B) (s : B ⟶ A) (sl : s ≫ f = 𝟙 B) (sr : s ≫ g = 𝟙 B) :
+    (WalkingReflexivePair ⥤ C) where
+  obj x :=
+    match x with
+    | zero => B
+    | one => A
+  map h :=
+    match h with
+    | WalkingReflexivePairHom.id _ => 𝟙 _
+    | left => f
+    | right => g
+    | reflexion => s
+    | right_reflexion => g ≫ s
+    | left_reflexion => f ≫ s
+  map_comp := by
+    rintro _ _ _ ⟨⟩ g <;> cases g <;>
+    simp only [Category.id_comp, Category.comp_id, Category.assoc, sl, sr] <;>
+    try {rfl}
+    iterate simp only [Category.id_comp, Category.comp_id, ← Category.assoc, sl, sr] <;> try {rfl}
+
+/-- (Noncomputably) bundle the data of a reflexive pair as a functor out of the walking reflexive
+pair -/
+noncomputable def ofIsReflexivePair (f g : A ⟶ B) [IsReflexivePair f g] :
+    (WalkingReflexivePair ⥤ C) :=
+  reflexivePair f g (commonSection f g) (section_comp_left f g) (section_comp_right f g)
+
+/-- The natural isomorphism between the diagram obtained by forgetting the reflexion of
+`ofIsReflexivePair f g` and the original parallel pair. -/
+noncomputable def inclusionWalkingReflexivePairOfIsReflexivePairIso
+    (f g : A ⟶ B) [IsReflexivePair f g] :
+    WalkingParallelPair.inclusionWalkingReflexivePair ⋙ (ofIsReflexivePair f g) ≅
+      parallelPair f g :=
+  diagramIsoParallelPair _
+
+end Limits
+
+namespace Limits
+
+open WalkingReflexivePair
+
+open WalkingReflexivePairHom
+
+variable {C : Type u} [Category.{v} C]
+
+@[simp]
+theorem reflexivePair_reflexion_comp_left (F : WalkingReflexivePair ⥤ C) :
+    F.map reflexion ≫ F.map left = 𝟙 F.obj zero :=
+  (F.map_id zero).symm.trans (F.map_comp reflexion left)|>.symm
```

Comment:
Using the `reflexion_comp_left` lemma suggested above: 
```suggestion
        F.map reflexion ≫ F.map left = 𝟙 (F.obj zero) := by
  rw [← F.map_comp, reflexion_comp_left, F.map_id]
```
(Note also the parentheses change.)